src/transport/zerotier/zerotier.c

```diff
@@ -1778,7 +1778,7 @@ zt_pipe_send(void *arg, nni_aio *aio)
 	fragsz = (uint16_t)(p->zp_mtu - zt_offset_data_data);
 
 	bytes = nni_msg_header_len(m) + nni_msg_len(m);
-	if (bytes >= (0xfffe * fragsz)) {
+	if ((uint16_t)bytes >= (0xfffe * fragsz)) {
```

Comment:
This needs to be size_t I think.  Or possibly just "unsigned".  The thing is that bytes might overflow a 16-bit value, which is why this check is here.

Reply:
Yeah, pretty sure we want to cast to (int) on the other side.  Because fragsz is the more constrained value.